benchmarks/strings/bench_strings.sh

```diff
@@ -20,375 +20,263 @@ source ../bench_plot.sh
+	bench_substring_variant "buffer" $1 $2 $3 $4 $5
 
-	../../../bin/nitc --global ../iteration_bench.nit
+	rm substr_bench
+}
 
-	prepare_res iter_flat_iter.out iter_flat_iter flatstring_iter
-	if $verbose; then
-		echo "FlatStrings by iterator :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats = $i, Loops = $3"
-		fi
-		bench_command $i flatstr_iter$i ./iteration_bench -m flatstr --iter-mode iterator --loops $2 --strlen $3 --ccts $i
+# $1: string or buffer
+# $2: loops
+# $3: strlen
+# $4: ccts min
+# $5: ccts increment
+# $6: ccts max
+function bench_substring_variant()
+{
+	tmp="${variant}_$1"
+	prepare_res out/substring/substring_$tmp.out $tmp $tmp
+	for i in `seq "$4" "$5" "$6"`; do
+		bench_command $i $tmp$i ./substr_bench -m $1 --loops $2 --strlen $3 --ccts $i
 	done
+}
 
-	prepare_res iter_flat_index.out iter_flat_index flatstring_index
-	if $verbose; then
-		echo "FlatStrings by index :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats = $i, Loops = $3"
-		fi
-		bench_command $i flatstr_index$i ./iteration_bench -m flatstr --iter-mode index --loops $2 --strlen $3 --ccts $i
-	done
+function bench_compiler()
+{
+	prepare_res out/compiler/compiler_$variant.out compiler_$variant compiler_$variant
 
-	prepare_res iter_buf_iter.out iter_buf_iter flatbuffer_iter
-	if $verbose; then
-		echo "FlatBuffers by iterator :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats = $i, Loops = $3"
-		fi
-		bench_command $i flatbuf_iter$i ./iteration_bench -m flatbuf --iter-mode iterator --loops $2 --strlen $3 --ccts $i
-	done
+	bench_command nitc nitc_$variant ../../bin/nitc ../../src/nitc.nit -o ../../bin/nitc
+}
 
-	prepare_res iter_buf_index.out iter_buf_index flatbuffer_index
-	if $verbose; then
-		echo "FlatBuffers by index:"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats = $i, Loops = $3"
-		fi
-		bench_command $i flatbuf_index$i ./iteration_bench -m flatbuf --iter-mode index --loops $2 --strlen $3 --ccts $i
+function basic()
+{
+	for i in lib_variants/*; do
+		variant=`basename "$i" | cut -f 1 -d '.'`
+		echo "---------------------------------------------------------"
+		echo "  Trying variant $variant for benchmark $bench"
+		echo "---------------------------------------------------------"
+		git am $i
+		../../bin/nitc ../../examples/hello_world.nit
+		./hello_world
+		git reset --hard $head
+		rm hello_world
 	done
+}
 
-	prepare_res iter_ropes_iter.out iter_ropes_iter ropes_iter
-	if $verbose; then
-		echo "Ropes by iterator :"
-	fi
-	for i in `seq 1 "$1"`; do
-		if $verbose; then
-			echo "String base length = $1, Concats (depth of the rope) = $i, Loops = $3"
-		fi
-		bench_command $i ropes_iter$i ./iteration_bench -m ropestr --iter-mode iterator --loops $2 --strlen $3 --ccts $i
-	done
+function prepare_compiler()
+{
+	cd ../../src
+	./ncall.sh
```

Comment:
I'm not a fan of regenerating nitc 3 times each time. I think that one should assume that the nitc in bin is fine